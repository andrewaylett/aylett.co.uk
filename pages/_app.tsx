import App from 'next/app'
import './global.scss'

export default App
