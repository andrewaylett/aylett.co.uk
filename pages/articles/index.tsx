import React from 'react'
import { Articles } from '../../articles'

export default function BlogList() {
  return (
    <>
      {Object.entries(Articles).map(([k, { metadata }]) => (
        <div key={metadata.title}>{metadata.title}</div>
      ))}
    </>
  )
}
