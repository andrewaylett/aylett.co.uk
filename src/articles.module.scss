.page {
    max-width: 50em;
    margin: auto;
}

.revision {
    font-size: smaller;
}

.revision::before {
    content: 'Â (';
}

.revision::after {
    content: ')';
}
