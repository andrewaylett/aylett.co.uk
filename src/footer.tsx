import React from 'react'

const Footer = () => <footer style={{ paddingTop: '1em' }}>Copyright Â© Andrew Aylett, 2020</footer>

export default Footer
