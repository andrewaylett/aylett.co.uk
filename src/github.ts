export const GITHUB_URL = (pageName: string): string =>
  `https://github.com/andrewaylett/aylett.co.uk/commits/main/src/pages${pageName.replace(
    /#.*/,
    ''
  )}.md`;
