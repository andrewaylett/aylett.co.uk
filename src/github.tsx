export const github_url = (page_name) =>
  `https://github.com/andrewaylett/aylett.co.uk/commits/main/src/pages${page_name}.mdx`
