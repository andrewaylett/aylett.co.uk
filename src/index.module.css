
.container {
    min-height: 100vh;
    padding: 0 0.5rem;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.container main {
    padding: 5rem 0;
    flex: 1;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.container footer {
    width: 100%;
    height: 100px;
    border-top: 1px solid #eaeaea;
    display: flex;
    justify-content: center;
    align-items: center;
}

.container footer img {
    margin-left: 0.5rem;
}

.container footer a {
    display: flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    color: inherit;
}

.title {
    text-align: center;
    margin: 0;
    line-height: 1.15;
    font-size: 4rem;
}

.description {
    text-align: center;
    line-height: 1.5;
    font-size: 1.5rem;
}

.logo {
    height: 1em;
}
