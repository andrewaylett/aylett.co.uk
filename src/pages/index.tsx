import Head from 'next/head'
import styles from '../index.module.scss'
import Link from 'next/link'
import React from 'react'

export const Home = (): JSX.Element => (
  <div className={styles.container}>
    <Head>
      <title>aylett.co.uk</title>
      <link rel="icon" href="/favicon.ico" />
    </Head>

    <main>
      <h1 className={styles.title}>Welcome to aylett.co.uk</h1>

      <p className={styles.description}>Insert &apos;90s &ldquo;site under construction&rdquo; gif here.</p>
      <p className={styles.description}>
        <Link href="/articles">Articles</Link>
      </p>
    </main>

    <footer>
      <a
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        target="_blank"
        rel="noopener noreferrer"
      >
        Powered by <img src="/vercel.svg" alt="Vercel Logo" className={styles.logo} />
      </a>
    </footer>
  </div>
)

export default Home
