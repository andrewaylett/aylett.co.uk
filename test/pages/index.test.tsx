import React from 'react'
import { render } from '../testUtils'
import { Home } from '../../src/pages'
import { describe, it, expect } from '@jest/globals'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
