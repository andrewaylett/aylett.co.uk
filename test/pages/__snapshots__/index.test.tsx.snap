// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <main>
      <h1
        class="title"
      >
        Welcome to aylett.co.uk
      </h1>
      <p
        class="description"
      >
        Insert '90s “site under construction” gif here.
      </p>
      <p
        class="description"
      >
        <a
          href="/articles"
        >
          Articles
        </a>
      </p>
    </main>
    <footer>
      <a
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by 
        <img
          alt="Vercel Logo"
          class="logo"
          src="/vercel.svg"
        />
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
